.circleci/config.yml

```diff
@@ -1,67 +0,0 @@
-version: 2
-jobs:
-  build:
-    docker:
-      - image: nixos/nix:2.0
-    environment:
-      GHC_WARNING_FLAGS:
-        "-Wall -Werror -Wcompat -Wincomplete-record-updates -Wincomplete-uni-patterns -Wredundant-constraints -Wnoncanonical-monad-instances"
```

Comment:
This drops `-Werror` and additional warnings on CI. Can you add those flags back? I guess the easiest is to just add them to the `.cabal` file. I think we had specified them on CI only because they can be annoying during development. But, during development one can also just disable them manually.

Reply:
On second thought, adding `-Werror ...` to the `.cabal` file could negatively impacting users trying to build `capability` on a different GHC version. Does `nixpkgs`'s `callCabal2nix` allow defining GHC options? Otherwise, maybe we could hide these behind a Cabal flag, e.g. `-fci`.

Reply:
Yes please, no `-Werror` in the `.cabal`.

Reply:
@regnat pointed me to the `configureFlags` attribute to Haskell `callPackage`. You could set `--ghc-option=-Wall` etc in there. I haven't found a way in which `callCabal2nix` exposes it, but `haskell.lib.overrideCabal` would allow to do that.

Reply:
I addressed this by adding `dev` flag which is only enabled on CI and disabled by default.